---
title: 'The Truth is Revealed: Identity of the Fish Fry "Teti" Fishery in Eastern Cuba'
author:
  - Sheila Rodriguez-Machado:
      email: srodr62@lsu.edu
      institute: [UofO]
      correspondence: true
  - Tomas M. Rodriguez-Cabrera:
      email: tomasmichelrodriguez@gmail.com
      institute: [UofA]
      correspondence: false
institute:
  - UofO: Museum of Natural Sciences, Louisiana State University
  - UofA: Sociedad Cubana de Zoologia
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    bookdown::word_document2:
      fig_caption: yes
      reference_docx: '../templates/template.docx'
      pandoc_args:
      - --lua-filter=../templates/scholarly-metadata.lua
      - --lua-filter=../templates/author-info-blocks.lua
      - --lua-filter=../templates/pagebreak.lua
bibliography: references.bib
csl: '../templates/journal-of-archaeological-science.csl'
abstract: |
  The lack of diagnostic characters to distinguish among species and even genera of gobies (Gobiidae) is still a challenge for science in the 21st century. However, the use of molecular tools applied to species-level identification has brought some light into the taxonomy and systematics of these fish taxa. The fish fry fishery in Eastern Cuba known as “Teti” represents a local fishery and there is no information about the exploitation of this natural resource outside this region. Fishing “Teti” has been a common practice for several years but an exponential increase of tourism in Eastern Cuba represents an additional threat to our biodiversity. Besides, if “Teti” refers to a single species or a species complex is still a knowledge gap in Cuba. The present study aims to identify the Fish Fry "Teti" Fishery in Eastern Cuba using molecular markers. We conducted a barcoding analysis comparing Cuban samples of gobies with samples of the genera *Sicydium* and *Awaous* available in GenBank. Here, we provide evidence that Busck´s Stone-Biting Goby, *Sicydium buscki*, is a component of the fish fry “Teti” fishery in Eastern Cuba. However, whether this fishery includes a single species or more than one species will remain as a mystery until more studies are conducted. Although this study represents the first effort toward and effective management, more thorough investigations and identification of the species comprising the “Teti” fishery are needed. It is imperative to include them in local management plans and to avoid overfishing this resource.  
keywords: |
  amphidromy; Busck's Stone-Biting Goby; COI; DNA barcoding; Gobiidae 
   
editor_options: 
  chunk_output_type: console
---

<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

<!-- With the following code you can access and display values from the yml header above. -->

Keywords: `r rmarkdown::metadata$keywords`

<!-- The following code chunk defines some general settings how code chunks should behave. -->

```{r setup, echo = FALSE}    
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  comment = "#>",
  fig.path = "../figures/",
  dpi = 300
)

#knitr::opts_knit$set(root.dir = '/rrcompendium/')

```


```{r}

# INSTALLING PACKAGES AND LIBRARIES - if needed
packages = c("BiocManager", "raster", "ggplot2", "ggthemes", "mapproj", "here",  
             "rentrez", "ape", "seqinr", "phangorn")

package.check <- lapply(packages, FUN = function(x) {
    if (!require(x, character.only = TRUE)) {
      install.packages(x, dependencies = TRUE)
      library(x, character.only = TRUE, include = FALSE, warn.conflicts = FALSE, quietly = TRUE)
    }
  }
)


if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
  
packagesBioconductor = c("msa", "Biostrings", "systemPipeR", "ggtree")
  
package.Bioconductor <- lapply(packagesBioconductor, FUN = function(y) {
    if (!require(y, character.only = TRUE)) {
      BiocManager::install(y, dependencies = TRUE)
      library(y, character.only = TRUE, warn.conflicts = FALSE, quietly = TRUE)
    }
  }
)


```

<!-- The actual document text starts here: -->

# Introduction

The family Gobiidae includes almost 2,000 species of marine, freshwater, and brackish fishes worldwide [@eschmeyer2020catalog]. This group is mostly represented by small-sized and bottom-dweller fishes found in tropical and subtropical waters. Such a small size along with cryptic body colors and the lack of diagnostic characters to distinguish among species and even genera is still a challenge for science in the 21st century. Only in the last ten years, more than a hundred gobiid species were described [@eschmeyer2020catalog]. However, the use of molecular tools applied to species-level identification has brought some light into the taxonomy and systematics of these fish taxa [@thacker2011phylogeny; @victor2014three; @keith2017schismatogobius]. 

Although most species in the family Gobiidae inhabit marine ecosystems, most species inhabiting freshwater or estuarine ecosystems display an amphidromous lifecycle [@myers1949usage; @bell1999overview]. This means that adults spawn in freshwater and when the eggs hatch, they are taken downstream by river flow into the ocean. Then, after spending from a few weeks to a few months (depending on the species), postlarvae return to rivers. This moment of postlarval returning is particularly fascinating because million of them gather at the river mouth as they start going upstream. This migratory behavior, particularly, when the postlarvae are going back to rivers, supports an artisanal fishery around the world [@bell1999overview; @mcdowall2007amphidromy]. Known as goby-fry fisheries, this is a common practice in several countries in the Indo-Pacific [@asis2013first], the Pacific [@castellanos2011goby], and the Caribbean [@neal2009factors]. The problem with it is that during the last years insufficient management of these fisheries has led to a considerable reduction in catches [@bell1999overview]. In addition, information on the identity of the species comprising these goby-fry fisheries in many places is still incomplete [@castellanos2011goby; @asis2013first]. 

Gobies have been poorly studied in Cuba and available information on their identification and distribution is not conclusive [@lara2010dna]. The fish fry fishery in Eastern Cuba known as “Teti” represents a local fishery and there is no information about the exploitation of this natural resource outside this region. Base on other reports in the Caribbean region [@neal2009factors], some authors claimed that the gobies *Sicydium plumieri* and *Sicydium punctatum* comprise the “Teti” fisheries in Cuba, but there is no solid evidence on that yet. Fishing “teti” has been a common practice for several years but an exponential increase of tourism in Eastern Cuba, due to recent economic policy changes, represents an additional threat to our biodiversity. Besides, if “teti” refers to a single species or a species complex is still a knowledge gap in Cuba. The present study aims to identify the Fish Fry "Teti" Fishery in Eastern Cuba using molecular markers.

# Methods

Tetis (19–28 mm standard length) were collected 400 m from the river mouth of Río Yumurí (Figure \@ref(fig:mainMap)), eastern Cuba, in December, 2018. Samples were collected using hand nets (1 mm mesh size) 2 m from the riverbank. Specimens were anesthetized using MS 222 (Tricaine Methanesulphonate) and immediately preserved in 80% ethanol. Voucher specimens were deposited in the collection of the Acuario Nacional de Cuba (catalog number: ANC-xxxx).

```{r mainMap, fig.cap="Sampling site at Yumuri river mouth, Eastern Cuba"}

# getting the map of Cuba
mydata <- getData("GADM", country = "cuba", level = 1)  # package raster
mymap <- fortify(mydata)                                # package ggplot2

# sampling site
dots <- data.frame(
  long = c(-74.293460),
  lat = c(20.298537),
  names = c("Yumuri river mouth"),
  stringsAsFactors = FALSE
  ) 

# main map
mainMap <-  ggplot() +
       geom_blank(data = mymap, aes(x = long, y = lat)) +
       geom_map(data = mymap, map = mymap, aes(group = group, map_id = id), fill = "gray90", color = "black", size = 0.3) +
       scale_x_continuous(limits = c(-74.51, -74.11), expand = c(0, 0)) +
       scale_y_continuous(limits = c(20.13, 20.42), expand = c(0, 0)) +
       geom_point(data = dots, aes(x = long, y = lat), color = "red", size = 4) +     
       coord_equal() +
       theme_bw()

# small square framing the zomm area
zoomOut <- data.frame(long = c(-74.49, -74.49, -74.09, -74.09, -74.49),
                   lat = c(20.13, 20.40, 20.40, 20.13, 20.13))

# inset map
insetMap <- ggplotGrob(
      ggplot() +
      geom_blank(data = mymap, aes(x = long, y = lat)) +
      geom_map(data = mymap, map = mymap, aes(group = group, map_id = id), fill = "gray95", color = "black", size = 0.3) +
      geom_path(data = zoomOut, aes(x = long, y = lat), size = 0.3) +
      scale_x_continuous(limits = c(-85.47, -73.65), expand = c(0, 0)) +
      scale_y_continuous(limits = c(19.5, 23.5), expand = c(0, 0)) +
      coord_map("polyconic") +
      theme_map() +     # package ggthemes
      theme(panel.background = element_rect(fill = NULL)) # fill = NULL or transparent, with color=NA both the background and the frame are transparent
      )

mainMap +
      annotation_custom(grob = insetMap, xmin = -74.12, xmax = -74.37,
                        ymin = 20.31, ymax = 20.42)


```

Total DNA was extracted from caudal peduncle muscle tissues using the DNAeasy tissue kit (QIAGEN). Partial sequences of the mitochondrial gene cytochrome oxidase I (COI) were amplified through the polymerase chain reaction in 20 μL of reaction volume (one unit of GoTaq DNA polymerase-Promega, 0.2 nM of each primer, 0.2 μM dNTPs, and 1.5 mM MgCl2) [@lara2010dna]. We used the primers COIf (5´-AAYCAYAAAGAYGGYACCCT-3´) and COIr (3´-CCTCNGGRTGNCCRAAGAAYCA-5´) [@palumbi1991simple].  

PCR products were purified using ExoSAP-IT™ PCR Product Cleanup Reagent (Affymetrix Inc./ USB, Cleveland, OH, USA), and then sequenced on an ABI 3100 automated sequencer (Applied Biosystems). Raw sequence files were edited and aligned with the software Geneious v.9.1 [@kearse2012geneious], and submitted to GenBank database (accession numbers #####). We used sequences of the genus *Sicydium* as reference for recovering the taxonomic identity of the "Tetis". Sequences of *Awaous banana* and *A. ocellaris* were used as outgroup. All non-Cuban sequences were recovered from GenBank using the R package *rentrez* [@winter2017rentrez]. Duplicated sequences and sequences with missing data were excluded. Sequences were edited using the R packages *seqinr* [@charif2007seqinr], *ape* [@paradis2015package], and *Biostrings* [@pages2013package]. Final alignment was obtained with the algorithm ClustalOmega [@sievers2011fast] using the R package *msa* [@bodenhofer2015msa].

```{r gettingAPI}

                                                            # DOWNLOADING SEQS FROM GENBANK
# rentrez: this package provides an R interface NCBI API and allows to search databases like GenBank (http://www.ncbi.nih.gov/genbank/) 

Sicydium_COI <- "Sicydium[Organism] AND COI[Gene]"

# I set retmax=580 because after running this by default, I got a message with the number of hits (580 in this case)
Sicydium_COI_search <- entrez_search(db = "nuccore", term = Sicydium_COI, retmax = 580)

# given that the search is too big for the system, it needs to be split into two smaller ones
Sicydium_COI_seqs1 <- entrez_fetch(db = "nuccore", id = Sicydium_COI_search$ids[0:295], rettype = "fasta")
Sicydium_COI_seqs2 <- entrez_fetch(db = "nuccore", id = Sicydium_COI_search$ids[296:580], rettype = "fasta")

# GETTING OUTGROUP SEQUENCES OF AWAOUS BANANA (2) AND A. OCELLARIS (2)

Awaous_bananaCOI <- "Awaous_banana[Organism] AND COI[Gene]"
Awaous_bananaCOI_search <- entrez_search(db = "nuccore", term = Awaous_bananaCOI, retmax = 2)
Awaous_bananaCOI_seqs <- entrez_fetch(db = "nuccore", id = Awaous_bananaCOI_search$ids, rettype = "fasta")

Awaous_ocellarisCOI <- "Awaous_ocellaris[Organism] AND COI[Gene]"
Awaous_ocellarisCOI_search <- entrez_search(db = "nuccore", term = Awaous_ocellarisCOI, retmax = 4)
Awaous_ocellarisCOI_seqs <- entrez_fetch(db = "nuccore", id = Awaous_ocellarisCOI_search$ids, rettype = "fasta")

# storing ALL sequences in a single fasta file
write(Sicydium_COI_seqs1, here(file="analysis/data/raw_data/Sicydium_COI_1-seqs.fasta"), sep = "\n")
# 'append = TRUE', I can keep adding seqs
write(Sicydium_COI_seqs2, here(file="analysis/data/raw_data/Sicydium_COI_1-seqs.fasta"), sep = "\n", append = TRUE)   
write(Awaous_bananaCOI_seqs, here(file="analysis/data/raw_data/Sicydium_COI_1-seqs.fasta"), sep = "\n", append = TRUE)
write(Awaous_ocellarisCOI_seqs, here(file="analysis/data/raw_data/Sicydium_COI_1-seqs.fasta"), sep = "\n", append = TRUE)


```


```{r editingAPI}

# seqinr: package for sequence manipulation 
# reading the file and renaming the sequences

Sicydium_COI_seqinrFormat <- read.fasta(here(file ="analysis/data/raw_data/Sicydium_COI_1-seqs.fasta"), seqtype = "DNA", as.string = TRUE, forceDNAtolower = FALSE)

Sicydium_COI_names <- attr(Sicydium_COI_seqinrFormat, "name")      # editing species name
Sicydium_COI_names <- gsub("\\..*", "", Sicydium_COI_names)        # eliminating characters after "." using gsub
Sicydium_COI_names <- gsub("^.*\\|", "", Sicydium_COI_names)       # eliminating characters before "|" using gsub

# using library "ape"
Sicydium_COI_apeFormat <- read.GenBank(Sicydium_COI_names)         # this line takes a few minutes to run

# creating a vector (to be added as name of the seqs) with the species, GenBank accession number, and type of gene
Sicydium_COI_GenBank_Ids <- paste(attr(Sicydium_COI_apeFormat, "species"), names(Sicydium_COI_apeFormat), sep = "_COI_")

# writing a new fasta file (2nd) using the vector above 
write.fasta(sequences = Sicydium_COI_seqinrFormat, names = Sicydium_COI_GenBank_Ids, nbchar = 10,
            here(file.out = "analysis/data/raw_data/Sicydium_COI_2-seqinrFormat.fasta"))


                                     # EDITTING SEQS AND GETTING THEM ALL READY FOR THE PHYLOGENETIC ANALYSIS

### REMOVING sequences of incomplete/unspecific NAMES; that is, those with 'sp.'
nameFasta <- read.fasta(here(file = "analysis/data/raw_data/Sicydium_COI_2-seqinrFormat.fasta"),
                        seqtype = "DNA", as.string = TRUE, forceDNAtolower = FALSE)

fastaNames <- strsplit(names(nameFasta), '_')                  # creating a vector splitting species names
completeSeqs <- which(! grepl("sp.", fastaNames))              # selecting only those with specific name in them  

# writing a new fasta (3rd) file with only those seqs with full scientific names
write.fasta(sequences = nameFasta[completeSeqs], names = names(nameFasta)[completeSeqs],
            here(file.out = "analysis/data/raw_data/Sicydium_COI_3-names.fasta"))

### REMOVING sequences of LENGTH shorter than 600 pb, given that the gene COI has between 627 and 658 bp.
lengthFasta <- read.fasta(here(file = "analysis/data/raw_data/Sicydium_COI_3-names.fasta"),
                          seqtype = "DNA", as.string = TRUE, forceDNAtolower = FALSE)     # 586 seqs
fastaLength <- lengthFasta[which(getLength(lengthFasta) >= 600)]
write.fasta(sequences = fastaLength, names = names(fastaLength),
            here(file.out = "analysis/data/raw_data/Sicydium_COI_4-lengths.fasta"))             # 401 seqs

### Removing DUPLICATE sequences
# using "seqinr" to do it 
# for not using attributes to name sequences
duplicatesFasta <- read.fasta(here("analysis/data/raw_data/Sicydium_COI_4-lengths.fasta"), set.attributes = FALSE) 

seqnames <- names(duplicatesFasta)
dups <- grep(TRUE,duplicated(duplicatesFasta))
uniqueNames <- seqnames[-dups]
uniqueSeqs <- duplicatesFasta[-dups] 
write.fasta(sequences = uniqueSeqs, names = uniqueNames,
            here(file.out = "analysis/data/raw_data/Sicydium_COI_5-dupli.fasta"))               # 204 seqs

# adding samples from Cuba (n=6)
mySamples <- read.fasta(here("analysis/data/raw_data/Sicydium_COI_Cuba.fas"))
write.fasta(mySamples, names(mySamples), here("analysis/data/raw_data/Sicydium_COI_5-dupli.fasta"), open = "a")              # 210 seqs


```


```{r alignment}
invisible({capture.output({                ### ALIGNING the sequences        # libraries "msa" and "Biostrings"

alignmentFasta <- readDNAStringSet(here("analysis/data/raw_data/Sicydium_COI_5-dupli.fasta"))
set.seed(123)
ClustalOAlignment <- msa(alignmentFasta, "ClustalOmega")
# print(ClustalOAlignment, showConsensus=FALSE, showNames=TRUE, show="complete") 

# Checking for the position of sequences that were not edited properly so they still have nucleotides different that A, G, C, T 
nucType <- alphabetFrequency(ClustalOAlignment)
unknownNuc <- nucType[, c(5:14)]  # this subset only contains nucleotides that are not A, G, T, C, or N or gaps
unknownPosition <- which(unknownNuc > 0, arr.ind=T)

# REMOVING samples with too many GAPS at the beginning and the end of the sequences and UNKNOWN nucleotides (ex.: W, K, S) 
colmask(ClustalOAlignment) <- IRanges(start=c(1,666),end=c(25,679))
rowmask(ClustalOAlignment) <- IRanges(start=c(1),end=c(1))
ClustalOAlignment2 <- as(ClustalOAlignment, "DNAStringSet")
writeXStringSet(ClustalOAlignment2, here(file="analysis/data/raw_data/Sicydium_COI_6-alignment.fasta"))            # 209 seqs -- 640 bp

### Replacing gaps at the begining and end with Ns
gapsFasta <- read.fasta(here(file = "analysis/data/raw_data/Sicydium_COI_6-alignment.fasta"), seqtype = "DNA", as.string = TRUE, forceDNAtolower = FALSE, apply.mask = FALSE)
#getLength(gapsFasta)
replacingGaps <- lapply(gapsFasta, function(x) gsub("-", "N", x))
write.fasta(replacingGaps, names(replacingGaps), here("analysis/data/raw_data/Sicydium_COI_7-gaps.fasta"))

})})

```


```{r, getRF, warning = FALSE}
invisible({capture.output({               ### GETTING the READING FRAME     --- library "systemPipeR"
                                          
alignedFasta <- readDNAStringSet(here("analysis/data/raw_data/Sicydium_COI_7-gaps.fasta")) 

getRF <- function(x){
  if (class(x)!= c("DNAStringSet")){
    stop("This function only takes files of class 'DNAStringSet'. Use package 'Biostrings' to read the input file")
  }
  if (any(letterFrequency(x, "-", as.prob=TRUE))>0){
    stop("This function do not take gaps as values/strings. Go and replaced gaps with Ns (if the gaps are at the begining or end of the sequences) or choose a subset of sequences that do not contain gaps")
  }
  orf1 <- predORF(x[1:10], n=1, type="df", mode="orf", strand="sense")   # predORF() do not read seqs with gaps
  # predicts longest ORF for sense strand in each query sequence                                                                          
  longestSeq1 <- orf1$start[orf1$width == max(orf1$width)]
  codonPosition1 <- longestSeq1[1]/3
  codonConfirmation1 <- round(codonPosition1)                         # 640 bp
  if (codonPosition1==codonConfirmation1){
   file.rename(here("analysis/data/raw_data/Sicydium_COI_7-gaps.fasta"), here("analysis/data/raw_data/Sicydium_COI_newRF.fasta")) 
  print("The Reading Frame of these seqs is +1, no further action is required, go to modelTesting")
    } else {
    if (codonPosition1!=codonConfirmation1){
      tempLength <- length(x[[1]])
      #remove first nucleotide position
      newRF2 <- subseq(x, start = 2, end = tempLength)      # 639 bp
      # getting new RF from the new alignment with -1 bp.
      orf2 <- predORF(newRF2[1:10], n=1, type="df", mode="orf", strand="sense") 
      # checking size gain
      longestSeq2 <- orf2$start[orf2$width == max(orf2$width)]
      codonPosition2 <- (longestSeq2[1]+2)/3
      codonConfirmation2 <- round(codonPosition2)
      if (codonPosition2==codonConfirmation2){
        writeXStringSet(newRF2, here(file="analysis/data/raw_data/Sicydium_COI_newRF.fasta"))                    
      print("The Reading Frame of these seqs was +2, new fasta file created, go to modelTesting")
      } else {
      if (codonPosition2!=codonConfirmation2){
        #remove first nucleotide position
        tempLength <- length(x[[1]])
        newRF3 <- subseq(x, start = 3, end = tempLength)   # 638 bp
        writeXStringSet(newRF3, here(file="analysis/data/raw_data/Sicydium_COI_newRF.fasta"))                                
      print("The Reading Frame of these seqs was +3, new fasta file created, go to modelTesting")
     }
   }  
  }
 }
}


suppressWarnings(suppressMessages(print(getRF(x = alignedFasta))))

})})


```

The nucleotide substitution model that best explained our data was estimated using the function modelTest from the R package *phangorn* [@schliep2011phangorn]. The phylogenetic tree was reconstructed following a maximum likelihood approach [@felsenstein1981evolutionary] using the functions optim.pml and bootstrap.pml, also from the package *phangorn*. The tree was edited and visualized using the R package *ggtree* [@yu2017ggtree].

This paper was completely developed in RStudio, PBC (“RStudio”) [@rstudio2020]. R code and data are available as a research compendium for generating this paper in the following link: https://github.com/sheilarm23/rrcompendium.git.    

```{r getBestModel, warning = FALSE}
                                      # FINDING THE BEST NUCLEOTIDE SUSBSTITUTION MODEL

# libraries ape and phangorn (modelTest is a function of the package phangorn)

fastaRF <- read.dna(here(file="analysis/data/raw_data/Sicydium_COI_newRF.fasta"), format = "fasta")

getBestModel <- function(y){
  if (class(y)!= c("DNAbin")){
    stop("This function only takes files of class 'DNAbin'. Use package 'ape' to read the input file")
  }
  formatPhyDat <- as.phyDat(y)                #getting the class used by modelTest: phyDat or pml
  set.seed(123)
  modelChoice <- modelTest(formatPhyDat, tree = NULL, model = c("JC", "K80", "HKY", "TrN", "SYM", "GTR"), multicore = TRUE, mc.cores = 4)
  modelChoiceSorted <- as.data.frame(modelChoice[order(modelChoice$AICc),])
  return(modelChoiceSorted)
}
  
bestModel <- suppressWarnings(suppressMessages(getBestModel(y = fastaRF)))               # getting the AICc for each competing model

# DISCLAIMER: visually inspecting the AICc is better than just getting the best model automatically, because it allows to compares models with similar AICc and it is my personal opinion to choose the one with the least parameters.  
# The alternative is to get automatically the first model that the data.frame modelTest gives and storing it as a variable. In order to get this as reproducible as possible, I chose the alternative.


```

```{r tree, echo = FALSE, include=TRUE, results = 'hide', fig.show='hide', warning = FALSE}

                                                        # BUILINDG THE TREE
# --- library: phangorn
set.seed(123)
env = attr(bestModel, "env")    # getting the AICc for each competing model sorted
set.seed(123)
fitStart = eval(get(bestModel$Model[which.min(bestModel$AICc)], env), env)    # initial fit for the model chosen

set.seed(123)
fit = suppressWarnings(optim.pml(fitStart, rearrangement = "stochastic", optGamma=TRUE, optInv=FALSE, model="HKY"))     # seeking a maximum likelihood solution
set.seed(123)    # getting the same tree topology each time the code runs 
bs = bootstrap.pml(fit, bs=100, optNni=TRUE, multicore=TRUE, mc.cores = 4)

# saving the tree in the folder 'analysis/figures/'
set.seed(123)
bsTree <- plotBS(midpoint(fit$tree), bs, p = 75, type="p", bs.col = "blue")     # fit$tree is where the optimal tree is stored

set.seed(123)
optimalTree <- plotBS(bsTree, type = "p", cex = 0.2, font = 1)
# export tree with bootstrap values as node labels
set.seed(123)
write.tree(optimalTree, here(file="analysis/data/derived_data/Sicydium_ML_tree.tre"))  


```


```{r loglikelihood, include = TRUE, results='hide'}
# this result wont be displayed in the final document
loglikelihood <- round(-1*(as.numeric(fit[1])), 2)

```


```{r editingTree, echo = FALSE}

# Editing and visualizing the tree to represent --- library: ggtree
tree <- read.tree(here(file="analysis/data/derived_data/Sicydium_ML_tree.tre"))

# checking that the tree is rooted
rootedTree <- is.rooted(tree)   # package ape. showing the tree root: geom_rootpoint(color="pink", size=5) 
# getting the node number for each main clade. This info is important because those #s are the one needed to edit each main clade/species
set.seed(123)
nodeNumber <- ggtree(tree) + geom_text(aes(label=node), hjust=-.5, col = "blue", size = 1) # looks horrible but it gives the clade number

```

# Results

The final alignment comprised 209 sequences, each one of 639 base pairs. Figure \@ref(fig:SicydiumTree) depicts the ML tree (-lnL = `r loglikelihood`) showing the relationships among all COI sequences identified as *Sicydium* spp. available in GenBank. Six clades matching known *Sicydium* species were recovered with strong support. The phylogenetic reconstruction shows that the six samples from Cuba, highlighted in green, nested within *Sicydium buscki* Evermann & Clark, 1906.


```{r SicydiumTree, fig.width = 15, fig.height = 19, fig.cap=paste("ML phylogenetic tree (-lnL =", loglikelihood,") depicting the relationships among all available COI sequences of the genus *Sicydium*. Sequences highlited in green represent samples from Cuba. Numbers on branches represent boostrap values higher than 82.")}

ggtree(tree, size = .8) + 
  xlim(-0.02, 0.26) +
  geom_treescale(x=0, y=-1.5, offset = -4, fontsize = 6, linesize = 0.8) + 
  geom_text2(aes(label=label, subset = !is.na(as.numeric(label)) & as.numeric(label) > 82), size = 6, col= "black", vjust = -0.3, hjust=1.9, na.rm = FALSE) + 
  geom_nodepoint(aes(label=label, subset = !is.na(as.numeric(label)) & as.numeric(label) > 82), size = 2, col= "black") + 
  geom_cladelabel(node=215, label="Sicydium plumieri", color="red", hjust = -0.07, fontsize = 6, offset = 0.002, fontface="italic") + 
  geom_cladelabel(node=108, label="Sicydium brevifile", color="blue", hjust = -0.008, vjust = 0.8, fontsize = 6, offset = 0.002, fontface="italic") + 
  geom_cladelabel(node=408, label="Sicydium salvini", color="purple", hjust = -0.07, fontsize = 6, offset = 0.002, fontface="italic") + 
  geom_cladelabel(node=320, label="Sicydium punctatum", color="brown", hjust = -0.07, fontsize = 6, offset = 0.002, fontface="italic") + 
  geom_cladelabel(node=371, label="Sicydium buscki", color="black", hjust = -0.07, fontsize = 6, offset = 0.002, fontface="italic") + 
  geom_cladelabel(node=407, label="Sicydium crenilabrum", color="orange", hjust = -0.03, vjust = 0.1, fontsize = 6, offset = 0.004, fontface="italic") +
  geom_cladelabel(node=211, label="Awaous spp.", color="darkgreen", hjust = -0.07, fontsize = 6, offset = 0.002, fontface="italic") +
  # samples Sicydium_COI_Cuba2,4,6 were nested within this clde but the distance is too short to be displayed in the tree
  geom_hilight(node=191, fill="green", alpha = 0.7, extend=.004) +
  geom_hilight(node=166, fill="green", alpha = 0.7, extend=.004) +
  geom_hilight(node=177, fill="green", alpha = 0.7, extend=.004) +
  geom_hilight(node=173, fill="green", alpha = 0.7, extend=.004) +
  geom_hilight(node=171, fill="green", alpha = 0.7, extend=.004) +
  geom_hilight(node=172, fill="green", alpha = 0.7, extend=.004) +
  theme(plot.margin = unit(c(0.5,0.5,0.5,0.5), "cm")) 

# suppressWarnings(suppressMessages(print(Sicydium_ML_tree)))


```

# Discussion

When it comes to nature conservation, species identification is a basic principle: what cannot be identified, will not be protected or conserved. Any successful management plan or effort will critically rely on a correct identification of the taxa intended to protect. However, when taxonomic issues exist, molecular tools have proved to be of considerable help uncovering largely hidden diversity [*e.g.*, @ardura2010dna; @kress2015dna]. Cuban freshwater fishes are an excellent example of the importance of applying molecular tools to species conservation, where more than ten new lineages have been discovered in the last years [@lara2010dna; @garcia2011molecular; @de2014phylogeography]. Nevertheless, some groups remain poorly studied, which prevents us from implementing appropriate conservation programs. 

Freshwater gobies (Gobiiformes, Gobiidae) have been a neglected group of fishes in Cuba. In fact, most information about their diversity and ecology on the island is virtually unknown. What we do know is that given their migratory life cycle, they have been the target of an artisanal fishery based on the postlarvae returning to rivers. This fish fry fishery, known as “Teti” fishery, is only practiced in the easternmost part of the island, and it has been reported from four rivers: Duaba, Yumuri, Toa, and Miel. The main problem with "Tetis", as it has been mentioned in many other countries harvesting postlarvae, is that species (and genus)-level identification is almost impossible in the field.   

Our results suggest that the “Teti” samples collected are *Sicydium buscki*. This is one of the four *Sicydium* species reported in the Caribbean islands [@eschmeyer2020catalog]. Although other species have been suggested to comprise this fishery in Cuba, this is the first report of any fish species based on molecular data. However, this study included a few samples from a single river. Other countries where fish fry fisheries also take place have reported several species of the order Gobiiformes [@castellanos2011goby; @asis2013first; @engman2019puerto]. Thus, additional surveys including more samples and rivers are needed to have a better idea, first of the diversity of gobies in Eastern Cuba, and second, of the fish diversity comprising the fish fry “Teti” fishery. Although harvest mainly occurs during the last-quarter moon phase from August to October [@tetiblogger2011], an increase in tourism in Cuba is leading to a higher demand for this natural resource. For this reason, further studies will contribute to our knowledge of Caribbean fish assemblages, which will allow us to better work toward their conservation and propose and develop effective management actions.   

# Conclusion

Busck´s Stone-Biting Goby, *Sicydium buscki*, is a component of the fish fry “teti” fishery in Eastern Cuba. This species represents the only taxa of the orden Gobiiformes confirmed by molecular tools as part of such a fishery.
More thorough investigations and identification of the species comprising the “teti” fishery are needed. It is imperative to include them in  local management plans and to avoid overfishing this resource.

# Acknowledgements

We thank the Habitat production team for their help during the field work. Sequencing was partially supported by a Smithsonian Tropical Research Institution Short-term Fellowship to SRM. In this sense, we thank advisement provided by Harilaos Lessios and the invaluable lab assistance of Eyda Gómez, Ligia and Axel Calderón. Additional lab work was also done with advice of Diego Elías. R code was developed with help of Tad Dallas, Heru Handika, and the community of Stack Overflow.   


<!-- The following line inserts a page break  -->

\newpage

# References 

<!-- The following line ensures the references appear here for the MS Word or HTML output files, rather than right at the end of the document (this will not work for PDF files):  -->

<div id="refs"></div>

\newpage

### Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies: 

```{r colophon, cache = FALSE}
# which R packages and versions?
if ("devtools" %in% installed.packages()) devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
if ("git2r" %in% installed.packages() & git2r::in_repository(path = ".")) git2r::repository(here::here())  
```
